fip-3.md

```diff
@@ -0,0 +1,158 @@
+---
+fip: 3
+title: Provide ability to cancel a request for funds
+status:Draft
+type: Functionality
+author: Ed Rotthoff <ed@dapix.io>
+created: 2020-04-16
+updated: 
+---
+
+## Abstract
+This FIP implements the following:
+* Adds new API end point and contract action for cancellation of a request for funds.
+* Adds a new endpoint which will support paging that is called get_cancelled_requests.
+* Adds new check to record obt action in the fio.request.obt contract to check that if the request id is specified, if there is a status of cancelled this is an error.
+
+
+## Motivation
+Presently the FIO API does not provide any way for a user to cancel a request for funds they have made:
+* It is foreseeable that users will have a desire to cancel newly made requests for funds
+* When a user enters errant data (for amounts or memos)
+* When events in the users world conspire to render the request irrelevant.
+* When a users account has been "hacked" and they wish to cancel fraudulent requests for funds.
+* Once we have an ability to cancel, it will be useful to have an api endpoint called get_cancelled_requests.
+* Once we have the ability to cancel, we want to check on record obt if the status of the request is cancelled, this is an error, whenever a request id is specified.
+
+
+## Specification
+### Cancel Funds Request
+#### New end point: *cancel_funds_request*
+#### New contract action canclrequest
+##### Request
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|fio_request_id|Yes|Positive Int|Existing Funds request id|
+|max_fee|Yes|Max fee SUFs|Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by /get_fee for correct value.|
+|tpid|Yes|FIO Address of TPID, See FIO Address validation rules|FIO Address of the wallet which generates this transaction. This FIO Address will be paid 10% of the fee.See FIO Protocol#TPIDs for details. Set to empty if not known.|
+|actor|Yes|the FIO account of the canceller|FIO pub account owning this payee FIO Address.|
+###### Example
+```
+{
+	"fio_request_id": "27",
+	"max_fee": 40000000000,
+	"tpid": ""
+    "actor":"edrtfgtrthg "
```

Comment:
check whitespace (shows a space at the end of the actor name) and is not lined up with the other json.